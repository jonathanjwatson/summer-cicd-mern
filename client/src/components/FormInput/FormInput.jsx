import React from "react";

const FormInput = ({ name, value, handleChange, id }) => {
  return (
    <div className="mb-3">
      <label htmlFor={id} className="form-label">
        {name}
      </label>
      <input
        type="text"
        className="form-control"
        id={id}
        name={id}
        value={value}
        onChange={(e) => handleChange(e.target.value)}
      />
    </div>
  );
};


export default FormInput;
