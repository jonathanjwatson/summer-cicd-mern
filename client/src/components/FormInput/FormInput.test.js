import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import FormInput from "./FormInput";

afterEach(cleanup);

describe("FormInput", () => {
  const props = {
    name: "Name",
    value: "",
    handleChange: jest.fn(),
    id: "restaurant-name",
  };

  it("calls handleChange on change", () => {
    const { getByLabelText } = render(<FormInput {...props} />);
    const nameInput = getByLabelText("Name");
    fireEvent.change(nameInput, { target: { value: "Waffle House" } });
    expect(props.handleChange).toHaveBeenCalled();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<FormInput {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
