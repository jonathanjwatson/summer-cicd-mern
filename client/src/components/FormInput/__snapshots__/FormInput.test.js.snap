// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="restaurant-name"
    >
      Name
    </label>
    <input
      class="form-control"
      id="restaurant-name"
      name="restaurant-name"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
